schulcloud/DashboardViewController.swift

```diff
@@ -24,37 +28,90 @@ final class DashboardViewController : UICollectionViewController {
 
     @IBOutlet var notificationBarItem : UIBarButtonItem!
 
+    lazy var noPermissionViewController : DashboardNoPermissionViewController = self.buildFromStoryboard(withIdentifier: "NoPermissionViewController")
+
     lazy var calendarOverview : CalendarOverviewViewController = self.buildFromStoryboard(withIdentifier: "CalendarOverview")
     lazy var homeworkOverview : HomeworkOverviewViewController = self.buildFromStoryboard(withIdentifier: "HomeworkOverview")
     lazy var notificationOverview = self.buildNotificationOverviewFromStroyboard()
 
-    lazy var viewControllers : [DynamicHeightViewController] = {
-        return [calendarOverview, homeworkOverview, notificationOverview]
-    }()
-
+    var viewControllers : [DynamicHeightViewController] = []
 
     var currentDesign : Design {
         return collectionView?.traitCollection.horizontalSizeClass == .regular ? .extended : .reduced
     }
 
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+        commonInit()
+    }
+
+    override init(collectionViewLayout layout: UICollectionViewLayout) {
+        super.init(collectionViewLayout: layout)
+        commonInit()
+    }
+
+    func commonInit() {
+        guard let currentUser = Globals.currentUser else { return }
+
+        if !currentUser.permissions.contains(.dashboardView) {
+            let missingPermissionController : DashboardNoPermissionViewController = self.buildFromStoryboard(withIdentifier: "NoPermissionViewController")
+            missingPermissionController.view.translatesAutoresizingMaskIntoConstraints = false
+            self.addChildViewController(missingPermissionController)
+            
+            missingPermissionController.missingPermission = .dashboardView
+            viewControllers.append(missingPermissionController)
+            return
+        }
+
+        var missingPermissions = [UserPermissions]()
+        if currentUser.permissions.contains(.calendarView) {
+            viewControllers.append(calendarOverview)
+            calendarOverview.view.translatesAutoresizingMaskIntoConstraints = false
+            self.addChildViewController(calendarOverview)
+        } else {
+            missingPermissions.append(.calendarView)
+        }
+
+        if currentUser.permissions.contains(.homeworkView) {
+            viewControllers.append(homeworkOverview)
+            homeworkOverview.view.translatesAutoresizingMaskIntoConstraints = false
+            self.addChildViewController(homeworkOverview)
+        } else {
+            missingPermissions.append(.homeworkView)
+        }
+
+        if currentUser.canDisplayNotification {
```

Comment:
Here you should check if the current design is extended

Reply:
The current is based on whether are not the horizontalSizeClass is compact or regular and is updated on rotation. Using this would break the landscape design on iPhone+ devices.